### PR TITLE
Implement backend status load in api v2

### PR DESCRIPTION
**Pull request by cosa65**
# Description
While implementing `getBackendStatus` I noticed that many of the functions of v1's implementation used functions in `general-services`, there's many things in there related to creating pipelines that use the same functions as the ones we are using.

It seemed easier for now and future work to move everything inside `general-serivces` in one go instead of only mving what we need for `getBackendStatus` since most of it doesn't care about dynamo/sql (it doesn't use it), mainly just the `getPipelineStatus` part which was updated to use sql.

This way it is ready to use when we implement runGem2s and qc for apiv2 and that part of the work will mainly be matter of checking that the implementations work well and connecting them to v2 endpoints.

Main changes I did while creating a copy of `general-service` but in v2:
- Change from kebabcase to camelcase filenames.
- Change some test descriptions.
- Change pipeline-manage to pipelineConstruct (let me know, I can change it to pipelineManage if that is less confusing).
- `getBackendStatus` was a utils function, that part was moved to the `experimentsController` and interacts with the helpers for `getPipelineStatus` and `getWorkerStatus`

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1834

https://ui-martinfosco-ui683-api332.scp-staging.biomage.net/
#### Link to staging deployment URL (or set N/A)
N/A

#### Links to any PRs or resources related to this PR
https://github.com/hms-dbmi-cellenics/ui/pull/683

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: src/api.v2/helpers/pipeline/getPipelineStatus.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    }).promise();
+  } catch (e) {
+    // if we get the execution does not exist it means we are using a pulled experiment so
+    // just return a mock sucess status
+    // TODO: state machines in production are deleted after 90 days, return a successful execution
+    // if the execution does not exist in production so the user will not be forced to re-run
+    // the pipeline losing annotations. This will be addressed checking if the
+    // processed files exist in S3 to avoid allowing users to move onwards when the pipeline was not
+    // actually run.
+    if ((e.code === pipelineConstants.EXECUTION_DOES_NOT_EXIST)
```
Comment (kafkasl) [inline src/api.v2/helpers/pipeline/getPipelineStatus.js]: instead of doing this, it would be great to save the result of 
```
 execution = await stepFunctions.describeExecution({
      executionArn,
    }).promise();
```
in the database so that when the data is removed 90 days after from aws we can query it.

File: src/api.v2/helpers/pipeline/getPipelineStatus.js [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+const pipelineSteps = getPipelineStepNames();
+
+// buildResponse function is wrapper function to ensure that all pipeline-status
+// responses contain the same information and parameters
+// more specific building response should rely on calling this one
+const buildResponse = (processName, execution, paramsHash, error, completedSteps) => {
+  const paramsHashOnlyIfGem2s = processName === pipelineConstants.GEM2S_PROCESS_NAME
+    ? paramsHash
+    : undefined;
+
```
Comment (ivababukova) [inline src/api.v2/helpers/pipeline/getPipelineStatus.js]: why was this addition necessary?

Comment (cosa65) [reply src/api.v2/helpers/pipeline/getPipelineStatus.js]: For qc all `paramsHash` are `null` (it's is what we get from sql), so I was replacing it with `undefined`, but it might make more sense and be simpler to just keep it `null` honestly. I'll remove this code.

Comment (cosa65) [reply src/api.v2/helpers/pipeline/getPipelineStatus.js]: Removed

**Review by ivababukova — APPROVED** [review]
